### PR TITLE
Adopt overlay/affix services

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**

### CODE REVIEW

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-adopt-overlay-affixer](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-adopt-overlay-affixer)

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-adopt-overlay-affixer](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-adopt-overlay-affixer)

File: e2e/autocomplete.e2e-spec.ts [hunk 1]
```diff
@@ -15,17 +16,17 @@ describe('Autocomplete', () => {
...
     input.value = 'r';
     input.click();
     browser.actions().sendKeys('r').perform();
 
-    browser.wait(() => {
-      return browser.isElementPresent(
-        element(by.css('.sky-dropdown-item'))
-      );
-    });
+    browser.wait(
```
Comment (Blackbaud-AlexKingman) [inline e2e/autocomplete.e2e-spec.ts:25]: Are all these waits here for the animation or loading results? Seems like we could mock those responses or kill the animations and not have to do any waiting. Is that possible?

File: src/app/public/modules/autocomplete/autocomplete-input.directive.ts [hunk 2]
```diff
@@ -97,14 +97,27 @@ export class SkyAutocompleteInputDirective implements OnInit, OnDestroy, Control
     return this.elementRef.nativeElement.value;
   }
 
-  public textChanges = new EventEmitter<SkyAutocompleteInputTextChange>();
-  public blur = new EventEmitter<void>();
+  public get textChanges(): Observable<SkyAutocompleteInputTextChange> {
+    return this._textChanges.asObservable();
+  }
+
+  public get blur(): Observable<void> {
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/autocomplete/autocomplete-input.directive.ts:104]: Check your alphabetical ordering here.

File: src/app/public/modules/autocomplete/autocomplete-input.directive.ts [hunk 3]
```diff
@@ -97,14 +97,27 @@ export class SkyAutocompleteInputDirective implements OnInit, OnDestroy, Control
...
+
+  private ngUnsubscribe = new Subject<void>();
+
+  private _blur = new Subject<void>();
 
-  private isFirstChange = true;
-  private ngUnsubscribe = new Subject();
   private _displayWith: string;
+
+  private _textChanges = new Subject<SkyAutocompleteInputTextChange>();
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/autocomplete/autocomplete-input.directive.ts:118]: It looks like we're not completing any of the `Subject`s in here.

File: src/app/public/modules/autocomplete/autocomplete.module.ts [hunk 4]
```diff
@@ -30,6 +28,7 @@ import {
 import {
   SkyLookupResourcesModule
 } from '../shared';
+import { SkyAutocompleteAdapterService } from './autocomplete-adapter.service';
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/autocomplete/autocomplete.module.ts:31]: Check your formatting here.

File: src/app/public/modules/autocomplete/autocomplete.component.html [hunk 5]
```diff
@@ -1,44 +1,50 @@
...
+  role="combobox"
+  [attr.aria-expanded]="isOpen"
+  [attr.aria-owns]="(isOpen) ? resultsListId : null"
+  [attr.id]="resultsWrapperId"
+>
   <ng-content>
   </ng-content>
-  <div class="sky-autocomplete-results">
-
-    <sky-dropdown
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/autocomplete/autocomplete.component.html:6]: Why remove the dropdown?

FYI- the `country-field-component` has some specific styles that are using dropdown selectors, and it looks like this change makes the flag on its own line:
Screen Shot 2020-04-06 at 11 43 38 AM

File: e2e/autocomplete.e2e-spec.ts (hunk 1 above)
Comment (Blackbaud-SteveBrush) [reply e2e/autocomplete.e2e-spec.ts:25]: They are waiting for the results to appear (there is a debounce time applied to the key events). This was rewritten from what was there before to accommodate the new overlays.

File: src/app/public/modules/autocomplete/autocomplete.component.html (hunk 5 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/autocomplete/autocomplete.component.html:6]: RE: "Why remove the dropdown?"
The intent of this pull request is to remove a needless dependency on the popovers module. The autocomplete has specific use-cases that don't apply to dropdown menus, etc.

Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/autocomplete/autocomplete.component.html:6]: Good catch with the country field styles.

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-adopt-overlay-affixer](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-adopt-overlay-affixer)

**Pull request merged**